packages/x-data-grid/src/material/index.tsx

```diff
@@ -120,6 +124,59 @@ const materialSlots: GridBaseSlots & GridIconSlotsComponent = {
 
 export default materialSlots;
 
+const CHECKBOX_COMPACT = { p: 0.5 };
+
+function BaseCheckbox(props: GridSlotProps['baseCheckbox'], ref: React.Ref<HTMLButtonElement>) {
+  const { autoFocus, label, fullWidth, slotProps, className, density, ...other } = props;
+
+  const elementRef = React.useRef<HTMLButtonElement>(null);
+  const handleRef = useForkRef(elementRef, ref);
+  const rippleRef = React.useRef<any>(null);
+
+  const sx = density === 'compact' ? CHECKBOX_COMPACT : undefined;
+
+  React.useEffect(() => {
+    if (autoFocus) {
+      const input = elementRef.current?.querySelector('input');
+      input?.focus({ preventScroll: true });
+    } else if (autoFocus === false && rippleRef.current) {
+      // Only available in @mui/material v5.4.1 or later
+      // @ts-ignore
+      rippleRef.current.stop({});
+    }
+  }, [autoFocus]);
+
+  if (!label) {
+    return (
+      <MUICheckbox
+        {...other}
+        className={className}
+        inputProps={slotProps?.htmlInput}
+        ref={handleRef}
+        sx={sx}
+        touchRippleRef={rippleRef}
+      />
+    );
+  }
+
+  return (
+    <MUIFormControlLabel
+      className={className}
+      control={
+        <MUICheckbox
+          {...other}
+          inputProps={slotProps?.htmlInput}
+          ref={handleRef}
+          sx={sx}
+          touchRippleRef={rippleRef}
+        />
+      }
+      label={label}
+      sx={fullWidth ? { width: '100%', margin: 0 } : undefined}
+    />
+  );
```

Comment:
@romgrk Hope it's okay to ask it directly here:
If I'm not mistaken I can't override the `baseCheckbox` slot in a good way anymore (without diving into the source code here).
I have to specifically recreate this conditional logic so that it works in the "Manage columns" overlay.
Or am I missing something?

Reply:
Wdym by override? What's the use-case?

Base slots are indeed expected to have a bit of friction to overwrite, these are meant for design-system authors to re-implement so they're more low-level than what the average end-user would use.

Reply:
Yes, sorry there's not way around the added complexity. We want the datagrid to be usable with other design-systems, which means material-ui needs to go through a standardized intermediate layer, the base typings. If all you care about is styling, you can still pass material props via the `material={{ ... }}` prop, possibly via `slotProps`. Otherwise, forking `/material/index.tsx` is best, it should be considered like the reference to implement custom bindings, either to material or to shadcn/mantine/etc.

Reply:
If it works for you, you could also wrap over the base checkbox instead of wrapping over the material-ui checkbox.

Reply:
I get your reasoning behind it but for me it gives the impression that other design-systems are "overvalued" compared to the native one.
From a pure consumer-perspective it should be easy to adjust the native components and hard to integrate another design-system.
Rough thought: Is it worth to make it harder for 99% of the people when 1% actually use another design-system.

This is highly conceptual, so I guess we can leave it here. 😅 
Thanks for the additional technical input, how to solve it. 👍